RHEL6_7/system/initscripts/ifcfg/preupg_network.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+orig_name_base="eth"
+new_name_base="net"
+conf_dir='/etc/sysconfig/network-scripts'
+temp_conf_dir='/root/preupgrade/dirtyconf/etc/sysconfig/network-configuration_fixed/'
+preupg_script='/root/preupgrade/preupgrade-scripts/rename_network.sh'
+udev_command='/sbin/udevadm info -a  -p'
+dev_path='/sys/class/net'
+index=0
+cat /dev/null > udev_temp 
+cat /dev/null > "$preupg_script"
+
+echo 'service network stop'  >> "$preupg_script"
+
+if ls "$dev_path" | grep -q "rename[0-9]+$";then
+    exit 1
+fi
+
+if grep -q "^MACADDR=" "$conf_dir"/ifcfg-*;then
+    exit 2
+fi
+
+net_indices=$(ls -1 "$dev_path" | egrep  "^net[0-9]+$"| wc -l)
+
+if ! [ -e "$temp_conf_dir" ];then
+    mkdir -p "$temp_conf_dir"
+fi
+
+for orig_full_name in $(ip a | awk -F': ' '/^[[:digit:]]/ {print $2}' | egrep "^eth[0-9]+$")
+do
+    orig_conf_file="${conf_dir}/ifcfg-${orig_full_name}"
+    mac=$(/sbin/ethtool -P $orig_full_name | awk '{ print $3}')
+    mac_rule=$(echo "ATTR{address}==\"$mac\"")
+    dev_type_rule=$($udev_command "${dev_path}/${orig_full_name}" | egrep "ATTR\{type\}")
+    index=$(echo $orig_full_name |sed -e "s/^$orig_name_base//g")
+    index=$(expr $net_indices + $index)
+    new_full_name="${new_name_base}$index"
+
+    if [ -f "$orig_conf_file" ];then
+        if [ -n "$new_full_name" ];then
+           new_conf_file="${temp_conf_dir}/ifcfg-${new_full_name}"
+           cp -p "$orig_conf_file" "$new_conf_file"
+           sed -i "s/^HWADDR=.*/HWADDR=$mac/g" "$new_conf_file"
+           sed -r -i "s/^NAME=.*|^DEVICE=.*/DEVICE=$new_full_name/g" "$new_conf_file"
+           echo "ip link set $orig_full_name name $new_full_name" >> "$preupg_script"
+           echo "cp -p $new_conf_file $conf_dir" >> "$preupg_script"
+           echo "rm -f $orig_conf_file" >> "$preupg_script"
+        fi
+    fi
+
+    echo -e "SUBSYSTEM==\"net\", ACTION==\"add\", DRIVERS==\"?*\", $mac_rule, $dev_type_rule, KERNEL==\"${orig_name_base}*\", NAME=\"$new_full_name\"\n" >> udev_temp
+
+done
+
+
+mv udev_temp "$temp_conf_dir"
+echo "cp -p $temp_conf_dir/udev_temp /etc/udev/rules.d/70-persistent-net.rules" >> "$preupg_script"
```

Comment:
The cp will overwrite the content of /etc/udev/rules.d/70-persistent-net.rules by the content of udev_temp. Is it expected? Shouldn't it append instead?

Reply:
No, this is an intended behaviour